src/chainlit/frontend/src/components/switch.tsx

```diff
@@ -0,0 +1,76 @@
+import { green } from 'palette';
+
+import MSwitch, { SwitchProps as MSwitchProps } from '@mui/material/Switch';
+import { styled } from '@mui/material/styles';
+
+type SwitchProps = {
+  checked?: boolean;
+  disabled?: boolean;
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+  onChange: (
+    event?: React.ChangeEvent<HTMLInputElement>,
+    checked?: boolean
+  ) => void;
+};
+
+export default function Switch(props: SwitchProps): JSX.Element {
+  const { checked, disabled, inputProps, onChange } = props;
+
+  return (
+    <StyledSwitch
+      disabled={disabled}
+      edge="end"
+      onChange={onChange}
+      checked={checked}
+      inputProps={inputProps}
+    />
+  );
+}
+
+const StyledSwitch = styled((props: MSwitchProps) => (
+  <MSwitch focusVisibleClassName=".Mui-focusVisible" disableRipple {...props} />
+))(({ theme }) => {
+  const isDarkMode = theme.palette.mode === 'dark';
+
+  return {
+    width: 40,
+    height: 24,
+    padding: 0,
+    '& .MuiSwitch-switchBase': {
+      margin: 0,
+      padding: '4px',
+      transitionDuration: '300ms',
+      '&.Mui-checked': {
+        transform: 'translateX(16px)',
+        color: '#fff',
+        '& + .MuiSwitch-track': {
+          backgroundColor: green[500],
+          opacity: 1,
+          border: 0
+        },
+        '&.Mui-disabled + .MuiSwitch-track': {
+          opacity: 0.5
+        }
+      },
+      '&.Mui-disabled .MuiSwitch-thumb': {
+        color: isDarkMode ? theme.palette.grey[600] : theme.palette.grey[100]
+      },
+      '&.Mui-disabled + .MuiSwitch-track': {
+        opacity: isDarkMode ? 0.3 : 0.7
+      }
+    },
+    '& .MuiSwitch-thumb': {
```

Comment:
the thumb should not have a box shadow (as in the figma)

Reply:
<img width="65" alt="Screenshot 2023-07-13 at 17 28 23" src="https://github.com/Chainlit/chainlit/assets/13104895/6e4a60cc-396c-493a-8afe-b5b9dde5192a">

Reply:
Good catch!